bigdata/src/main/scala/org/w3/banana/bigdata/BigdataOps.scala

```diff
@@ -0,0 +1,135 @@
+package org.w3.banana.bigdata
+
+import java.util.{Locale, NoSuchElementException}
+
+import com.bigdata.rdf.model._
+import org.openrdf.model.Resource
+import org.openrdf.model.impl._
+import org.w3.banana.bigdata.Bigdata
+import org.w3.banana.isomorphism.{GraphIsomorphism, SimpleMappingGenerator, VerticeCBuilder}
+import org.w3.banana.{DefaultURIOps, RDFOps}
+
+class BigdataOps(implicit config:BigdataConfig[Bigdata]) extends RDFOps[Bigdata] with DefaultURIOps[Bigdata] with BigdataMGraphOps
+{
+
+  lazy val valueFactory: BigdataValueFactory= BigdataValueFactoryImpl.getInstance(config.basePrefix)
+
+  def emptyGraph: Bigdata#Graph = BigdataGraph(Map.empty,0)
+
+  def makeGraph(triples: Iterable[Bigdata#Triple]): Bigdata#Graph =    triples.foldLeft(emptyGraph) { (g, triple) => g + triple }
+
+  def getTriples(graph: Bigdata#Graph): Iterable[Bigdata#Triple] = graph.triples
+
+  // triple
+
+  def makeTriple(s: Bigdata#Node, p: Bigdata#URI, o: Bigdata#Node): Bigdata#Triple =
+    s match {
+      case res:Resource=>  valueFactory.createStatement(res,p,o)
+
+      case _=> throw new RuntimeException("makeTriple: in Sesame subject " + p.toString + " must be a either URI or BlankNode")
+    }
+
+  def fromTriple(t: Bigdata#Triple): (Bigdata#Node, Bigdata#URI, Bigdata#Node) =  (t.getSubject, t.getPredicate, t.getObject)
+
+  def foldNode[T](node: Bigdata#Node)(funURI: Bigdata#URI => T, funBNode: Bigdata#BNode => T, funLiteral: Bigdata#Literal => T): T = node match {
+    case iri: Bigdata#URI => funURI(iri)
```

Comment:
I know this is done like that in other places... But is there a visitor in bigdata? If yes, you should really use that. (errors are just too easy with types otherwise)